docs/spec.md

```diff
@@ -1,104 +1,87 @@
 
 
-## Type
+## Module
 
-A type in Static Land is a dictionary (JavaScript object) with static functions as values.
-'Static' means that functions don't use `this`, they can be detached from the type object.
-The type object is just a container for functions.
+Module is a JavaScript object, that contains some static functions and values.
+Static means that functions don't use `this`, and when detached from the object work in the same way.
+Here is an example:
 
 ```js
-const {of, map} = MyType
-
-// This should work
-map(x => x + 1, of(41)) // MyType(42)
-```
-
-Functions from type object are often called "methods" of the type.
-But keep in mind that they are not "methods" in JS sense (they don't use `this`).
-
-## Type signatures
-
-Each method in this spec comes with a type signature, that looks like the following.
-
-```
-map :: Functor f => Type f ~> (a → b, f a) → f b
+const FooModule = {
+  foo: 1,            // a value
+  bar: (x) => x + 1, // a function
+}
 ```
 
-We use syntax similar to Haskell's. You can learn about it from
-[Ramda's wiki](https://github.com/ramda/ramda/wiki/Type-Signatures) or from the book
-["Professor Frisby's Mostly Adequate Guide to Functional Programming"](https://drboolean.gitbooks.io/mostly-adequate-guide/content/ch7.html).
+Note that this has nothing to do with JavaScript module systems like ES6 modules,
+in this specification a module is just an object.
 
-This spec uses the following extensions to the type signature syntax:
 
-  1. `(a, b) → c` denotes a binary function which is not curried. Same for more arguments.
-  1. `Type a` denotes the [type dictionary](#type) of the type `a`.
-     For instance a function with a signature `(Type f, f a) → f a` can be called as `fn(F, F.of(1))`.
-  1. `~>` denotes a property access on a JavaScript object.
-     For example `fn :: Type f ~> (f a) → f a` can be applied as `F.fn(F.of(1))`.
+## Module Signature
 
-If a method called with incorrect types the behaviour is unspecified.
-Also if a method accepts a function it must only apply the function in accordance with
-the type signature (i.e. provide the correct number of arguments of the correct types).
+Module signature describes an interface that a module can match. The syntax is very similar to
+that of Flow or TypeScript. Here is an example of a signature that the `FooModule` above matches:
 
-
-## Parametricity
-
-All methods' implementations should only use type information about arguments that is known from the
-methods' type signatures. It's not allowed to inspect arguments or values that they produce
-or contain to get more information about their types. In other words methods
-should be [parametrically polymorphic](https://en.wikipedia.org/wiki/Parametric_polymorphism).
-
-For example let's take a look at Functor's `map` signature:
-
-```
-map :: Functor f => Type f ~> (a → b, f a) → f b
+```js
+Foo {
+  foo: number,
+  bar: (number) => number
+}
```

Comment:
Bottom line with the signatures: I want to keep them simple and don't put too much information into them. Complex stuff easier to write in english.